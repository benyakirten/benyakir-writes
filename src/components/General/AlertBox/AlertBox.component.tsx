import * as React from 'react'

import { StyledAlertBox } from './AlertBox.styles'

import { AlertBoxProps } from '@Types/props';

const AlertBox: React.FC<AlertBoxProps> = ({ success = true, children }) => {
    return (
        <StyledAlertBox success={success}>
            {children}
        </StyledAlertBox>
    )
}

export default AlertBox;