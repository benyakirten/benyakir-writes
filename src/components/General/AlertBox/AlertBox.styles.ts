import styled from "styled-components"

import { rollOut } from "@Styles/animations"
import { FONT_MD } from "@StyleVars"

export const StyledAlertBox = styled.div<{ success: boolean }>`
    position: relative;

    display: flex;
    justify-content: center;
    align-items: center;

    padding: 2rem 3rem;
    
    background-color: ${props => props.success ? props.theme.alertBox.alert.success : props.theme.alertBox.alert.error};
    color: ${props => props.theme.alertBox.textColor};
    font-size: ${FONT_MD};
    border: 2px solid ${props => props.theme.base.border};

    transform-origin: top;

    animation: ${rollOut} 1s ease forwards;
`