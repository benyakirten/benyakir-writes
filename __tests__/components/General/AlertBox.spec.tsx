import * as React from "react";
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import {
    cleanup,
    render
} from "@testing-library/react";

import AlertBox from "@Gen/AlertBox/AlertBox.component"

describe('AlertBox component', () => {
    it('should render successfully', () => {
        expect(() => render(<AlertBox />)).not.toThrow()
        cleanup()
    })

    it('should have a green or red background if the success prop is true or false', () => {
        const comp = renderer.create(<AlertBox success={true} />)
        expect(comp.toJSON()).toMatchSnapshot()
        expect(comp.toJSON()).toHaveStyleRule("background-color", "#44764d")

        comp.update(<AlertBox success={false} />)
        expect(comp.toJSON()).toMatchSnapshot()
        expect(comp.toJSON()).toHaveStyleRule("background-color", "#993d4d")
    })

    it('should have a default success prop value of true', () => {
        const comp = renderer.create(<AlertBox />)
        expect(comp.toJSON()).toMatchSnapshot()
        expect(comp.toJSON()).toHaveStyleRule("background-color", "#44764d")
    })
})