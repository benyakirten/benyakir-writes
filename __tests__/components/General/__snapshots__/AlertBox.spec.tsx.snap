// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox component should have a default success prop value of true 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem 3rem;
  background-color: #44764d;
  color: #FFF;
  font-size: 1.4rem;
  border: 2px solid #000;
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-animation: YsjJA 1s ease forwards;
  animation: YsjJA 1s ease forwards;
}

<div
  className="c0"
/>
`;

exports[`AlertBox component should have a green or red background if the success prop is true or false 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem 3rem;
  background-color: #44764d;
  color: #FFF;
  font-size: 1.4rem;
  border: 2px solid #000;
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-animation: YsjJA 1s ease forwards;
  animation: YsjJA 1s ease forwards;
}

<div
  className="c0"
/>
`;

exports[`AlertBox component should have a green or red background if the success prop is true or false 2`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem 3rem;
  background-color: #993d4d;
  color: #FFF;
  font-size: 1.4rem;
  border: 2px solid #000;
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-animation: YsjJA 1s ease forwards;
  animation: YsjJA 1s ease forwards;
}

<div
  className="c0"
/>
`;
